.buildkite/scripts/e2e-pipeline/bootstrap.py

```diff
@@ -0,0 +1,165 @@
+"""
+E2E bootstrapping with Python script
+    - A script to resolve stack versions
+    - Download and spin up elastic-package
+    - Clone integrations repo and prepare packages
+    - When E2E finishes, teardown the stack
+"""
+import os
+import sys
+import time
+import util
+
+
+class Bootstrap:
+    ELASTIC_PACKAGE_DISTRO_URL = "https://api.github.com/repos/elastic/elastic-package/releases/latest"
+    LOGSTASH_CONTAINER_NAME = "elastic-package-stack-e2e-logstash-1"
+    PLUGIN_NAME = "logstash-filter-elastic_integration"
+
+    def __init__(self, stack_version: str, project_type: str) -> None:
+        f"""
+        A constructor of the {Bootstrap}.
+
+        Args:
+            stack_version: An Elastic stack version where {Bootstrap} spins up with
+            project_type: type of the project running with, example serverless or on-prems
+
+        Returns:
+            Validates and sets stack version, project type and resolves elastic package distro based on running OS (sys.platform)
+        """
+        self.stack_version = stack_version
+        self.__validate_and_set_project_type(project_type)
+        self.__resolve_distro()
+
+    def __validate_and_set_project_type(self, project_type: str):
```

Comment:
```suggestion
    def __validate_and_set_project_type(self, project_type: str) -> None:
```
nit: just to match the rest of the return type spec in this file :)